internal/codegen/primitive.go

```diff
@@ -11,16 +11,17 @@ import (
 type PrimitiveType struct {
 	Type        string
 	newInstance func() interface{}
+	empty       interface{}
 }
 
 var PrimitiveTypes = []PrimitiveType{
-	{Type: "int32", newInstance: func() interface{} { return new(int32) }},
-	{Type: "int64", newInstance: func() interface{} { return new(int64) }},
-	{Type: "float32", newInstance: func() interface{} { return new(float32) }},
-	{Type: "float64", newInstance: func() interface{} { return new(float64) }},
-	{Type: "bool", newInstance: func() interface{} { return new(bool) }},
-	{Type: "string", newInstance: func() interface{} { return new(string) }},
-	{Type: "bytes", newInstance: func() interface{} { return new([]byte) }},
+	{Type: "int32", newInstance: func() interface{} { return new(int32) }, empty: int32(0)},
+	{Type: "int64", newInstance: func() interface{} { return new(int64) }, empty: int64(0)},
+	{Type: "float32", newInstance: func() interface{} { return new(float32) }, empty: float32(0.0)},
+	{Type: "float64", newInstance: func() interface{} { return new(float64) }, empty: float64(0.0)},
+	{Type: "bool", newInstance: func() interface{} { return new(bool) }, empty: false},
+	{Type: "string", newInstance: func() interface{} { return new(string) }, empty: ""},
+	{Type: "bytes", newInstance: func() interface{} { return new([]byte) }, empty: nil},
```

Comment:
This might be kind of a weird nit but I feel like explicitly initializing stuff to the 0 value is hairy. Might be the Java in me talking, and it's obviously _very_ unlikely that the 0-value for anything will change, but it's the kind of stuff that bugs me when I see it. I'm not going to hold you to it, but I would have written it like this: ```*new(int32)```, just in case